src/containerConfig.ts

```diff
@@ -1,49 +1,121 @@
-    {
-      token: 'onSignal',
-      provider: {
-        useValue: {
-          useValue: async (): Promise<void> => {
-            await Promise.all([tracing.stop(), metrics.stop()]);
-          },
+    { token: SERVICES.METRICS, provider: { useValue: metrics } },
+  ];
+}
+
+async function getK8sDependencies(shutdownHandler: ShutdownHandler): Promise<InjectionObject<unknown>[]> {
+  const jobLabels: Record<string, string> = {
+    app: 'job-chief',
+    'owner-id': config.get('app.instanceUid'),
+    environment: process.env.NODE_ENV ?? 'development',
+  };
+
+  const [kubeConfig, k8sApi, k8sJobApi] = createK8sApis(config.get<boolean>('kubernetes.loadConfigFromCluster'));
+
+  const jobInformer = createJobInformer(kubeConfig, k8sJobApi, config.get<string>('kubernetes.namespace'), jobLabels);
+
+  await jobInformer.start();
+  shutdownHandler.addFunction(jobInformer.stop.bind(jobInformer));
+
+  return [
+    { token: JOB_LABELS_SYMBOL, provider: { useValue: jobLabels } },
+    { token: SERVICES.K8S_CONFIG, provider: { useValue: kubeConfig } },
+    { token: SERVICES.K8S_API, provider: { useValue: k8sApi } },
+    { token: SERVICES.K8S_JOB_API, provider: { useValue: k8sJobApi } },
+    { token: SERVICES.K8S_JOB_INFORMER, provider: { useValue: jobInformer } },
+    { token: SERVICES.K8S_JOB_FACTORY, provider: { useFactory: instancePerContainerCachingFactory(jobFactoryForDi) as FactoryFunction<unknown> } },
+  ];
+}
+
+async function getJobsConfig(): Promise<JobConfig[]> {
+  const jobsConfigRaw = await readFile(config.get('app.jobsConfigPath'), 'utf8');
+  const jobsConfig = validateJobConfig(JSON.parse(jobsConfigRaw));
+  return jobsConfig;
+}
+
+export interface RegisterOptions {
+  override?: InjectionObject<unknown>[];
+  useChild?: boolean;
+}
+
+export const registerExternalValues = async (options?: RegisterOptions): Promise<DependencyContainer> => {
+  const shutdownHandler = new ShutdownHandler();
+  try {
+    const jobsConfig = await getJobsConfig();
+
+    const pgBoss = await pgBossFactory(config.get<DbConfig>('db'));
+
+    const k8sDeps = await getK8sDependencies(shutdownHandler);
+    const obsDeps = getObservabilityDependencies(shutdownHandler);
+
+    const dependencies: InjectionObject<unknown>[] = [
+      { token: ShutdownHandler, provider: { useValue: shutdownHandler } },
+      { token: SERVICES.CONFIG, provider: { useValue: config } },
+      ...obsDeps,
+      ...k8sDeps,
+      { token: JOBS_CONFIG_SYMBOL, provider: { useValue: jobsConfig } },
+      { token: PgBoss, provider: { useValue: pgBoss } },
+      {
+        token: SERVICES.JOB_MANAGER_FACTORY,
+        provider: { useFactory: instancePerContainerCachingFactory(jobManagerFactoryForDi) as FactoryFunction<unknown> },
+      },
+      {
+        token: QUEUE_PROVIDER_SYMBOL,
+        provider: { useClass: PgBossQueueProvider },
+        options: { lifecycle: Lifecycle.Singleton },
```

Comment:
decoration on class level is more straightforward

Reply:
because we register as interface its not possible to use decorator